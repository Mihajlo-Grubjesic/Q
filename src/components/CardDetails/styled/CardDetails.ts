import styled from 'styled-components';

export const CardDetailsWrapper = styled.div`
  display: flex;
  height: 700px;
  background: ${({ theme }) => theme.background};
  border-radius: 4px;
  color: ${({ theme }) => theme.color};
  flex-direction: column;
  align-items: center;
  box-shadow: ${({ theme }) => ` 0px 0px 2px 1px ${theme.color}`};
  padding: 16px;
  margin-bottom: 8px;
`;
