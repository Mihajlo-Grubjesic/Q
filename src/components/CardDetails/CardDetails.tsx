import { CardSubtitle, CardTitle } from "../Card/styled/Card";
import { CardDetailsWrapper } from "./styled/CardDetails";

interface CardDetailsProps {
  title?: string;
  subtitle?: string;
  content?: JSX.Element;
}

export const CardDetails = ({
  title = "",
  subtitle = "",
  content,
}: CardDetailsProps): JSX.Element => {
  return (
    <>
      <CardDetailsWrapper>
        <CardTitle>{title}</CardTitle>
        <CardSubtitle>{subtitle}</CardSubtitle>
        {content}
      </CardDetailsWrapper>
    </>
  );
};
