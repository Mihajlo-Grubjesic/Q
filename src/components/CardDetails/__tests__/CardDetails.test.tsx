import { render, screen } from '@testing-library/react';
import { CardDetails } from '../CardDetails';

describe('CardDetails', () => {
  it('renders card details', () => {
    const { container } = render(
      <CardDetails title="title" subtitle="subtitle" />,
    );
    expect(screen.getByText('title')).toBeInTheDocument();
    expect(screen.getByText('subtitle')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
