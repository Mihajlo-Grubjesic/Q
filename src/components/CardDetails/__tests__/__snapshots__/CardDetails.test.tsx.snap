// Vitest Snapshot v1

exports[`CardDetails > renders card details 1`] = `
<div>
  <div
    class="sc-jSUZER pdkVY"
  >
    <div
      class="sc-hLBbgP kaMPmA"
    >
      title
    </div>
    <div
      class="sc-eDvSVe cwgtvh"
    >
      subtitle
    </div>
  </div>
</div>
`;
